src/browser/browser.ts

```diff
@@ -561,6 +563,14 @@ export class Browser {
     this.log.debug('Browser request', 'url', req.url(), 'method', req.method());
   };
 
+  logRedirectResponse = (resp: puppeteer.HTTPResponse) => {
+    const status = resp.status()
+    if (status >= 300 && status <= 399) {
+      const headers = resp.headers();
+      this.log.debug(`Redirect from ${resp.url()} to ${headers['location']}`)
```

Comment:
It seems like one redirect request contains this `Location` header while the others do not, leading to a few `Redirect from xxx to undefined` logs.
```suggestion
      const redirectLocation = headers['location'];
      if (redirectLocation) {
        this.log.debug(`Redirect from ${resp.url()} to ${redirectLocation}`);
      }
```

Reply:
why are there redirects without Location header? Shouldn't those be also logged to understand the flow?